### PR TITLE
Feature#16/admin authentication

### LINKED ISSUE #16: Validate that an administrator exists and display an error message for incorrect credentials
# REQUIREMENTS
It must validate the information provided by the admin and show a confirmation message.
- Check if the input contains the needed information
- Search in database the information to validate
- Show a message after validation.

# DESIGN
image

### PR DESCRIPTION
**Pull request by nallely64**
# Description
 - What?: Implementation of an alert message for incorrect credentials for the admin login.
 - Why?: To give feedback to the user.
 For more info review issue #16 

  ## Screenshots
Screenshot from 2022-03-24 19-13-54

### CODE REVIEW

File: app/javascript/application.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
 // Configure your import map in config/importmap.rb. Read more: https://github.com/rails/importmap-rails
+import 'bootstrap'
 import "@hotwired/turbo-rails"
 import "controllers"
+//= require jquery3
+//= require popper
+//= require bootstrap
```
Comment (YilKanda) [inline app/javascript/application.js]: Add a blank line

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,7 @@
 #
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
+# StreetCp.create(street_name: "Calle", cp: "28000")
+# Branch.create(name: "Calle", street_cp_id: 1, city: "Colima", town: "Villa De Alvarez", number: 955555555)
+# Employee.create(name: "Juan", email: "juan@gmail.com", position: "Gerente", private_number: 123456789, branch_id: 1)
+# Admin.create(email: "admin@admin.com", encrypted_password: "admin123")
```
Comment (YilKanda) [inline db/seeds.rb]: If you are not going to occupy this section of code, instead of commenting it out, it should be removed, to keep your code clean.

Comment (AlejandroCuriel) [reply db/seeds.rb]: thanks, done :100:

File: app/javascript/application.js (hunk 1 above)
Comment (AlejandroCuriel) [reply app/javascript/application.js]: thanks, done :star:

**Review by diegoOG09 — DISMISSED** [review]
LGTM

File: db/migrate/20220323232155_devise_create_admins.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
```
Comment (JuanequeX) [inline db/migrate/20220323232155_devise_create_admins.rb]: If this code are not necessary, please delete it

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -12,6 +12,13 @@
 
   <body>
     <%= render 'layouts/header' %> 
+    <% if notice%>
+      <%= javascript_tag "alert('#{flash[:notice]}')"%>
+    <%end%>
+    <%if alert%>
+     <%= javascript_tag "alert('#{flash[:alert]}')"%>
```
Comment (YilKanda) [inline app/views/layouts/application.html.erb]: Check your indentation

Comment (germanruzca) [reply app/views/layouts/application.html.erb]: Done, thanks

File: db/migrate/20220323232155_devise_create_admins.rb (hunk 3 above)
Comment (germanruzca) [reply db/migrate/20220323232155_devise_create_admins.rb]: Done, thanks

**Review by Javi1499 — APPROVED** [review]
Good job!

**Review by EliasUR — APPROVED** [review]
lgtm <3

**Review by VictorElizalde — APPROVED** [review]
LGTM, just be careful with the forms you are using, you can use simple-form gem for an easier sintax

**Pull request merged**